setup.py

```diff
@@ -92,6 +92,7 @@ def build_extension(self, ext):
         "pymbar>3.0.4",
         "pyyaml",
         "scipy",
+        "mypy",
```

Comment:
Could you comment on why this change was needed?

Reply:
Because of the change to require `stubgen` to build the package, you can't install this without mypy based on this comment (https://github.com/proteneer/timemachine/pull/711#discussion_r848956244)

We could go back to an explicit version of stubgen and keep mypy as a separate requirement. We probably should add some form of testing in CI that doing `pip install .` works out of the box, as right now it doesn't.

Reply:
After digging in a bit more with @mcwitt it looks like we have a discontinuity between `make build` and `pip install`. `pip install` reaches into `pyproject.toml` and gets the build requirements there (that include mypy) and builds just fine, however `make build` will fail as `mypy` isn't in the env. 

Probably warrants an offline chat on how to proceed

Reply:
Just a note that if we decide to leave things as-is, we should probably add a note to the readme that `make build` only works after installing the dev dependencies (`pip install .[dev]`)

Reply:
ah I see! I'm okay either way. (I do make build)